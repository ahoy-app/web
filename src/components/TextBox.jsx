import React from 'react'
import Radium from 'radium'
import PropTypes from 'prop-types'

const styles = {
  backgroundColor: 'white',
  height: '40px',
  // width: '100%',
  flex: 1,

  borderRadius: '1rem',
  border: 'none',
  filter: 'drop-shadow(2px 5px 2px #0003)',

  padding: '0px 10px 0px 10px',
  boxSizing: 'border-box',

  lineHeight: '2rem',
  fontSize: '1em',
  fontFamily:
    '-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
}

const TextBox = ({ value, onChange }) => (
  <input style={styles} value={value} onChange={onChange} />
)

TextBox.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
}

export default Radium(TextBox)
