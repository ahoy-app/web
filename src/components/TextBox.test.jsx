import React from 'react'
import { shallow } from 'enzyme'

import TextBox from './TextBox'

describe('TextBox', () => {
  it('should render correctly', () => {
    const component = shallow(<TextBox />)
    expect(component).toMatchSnapshot()
  })
})
