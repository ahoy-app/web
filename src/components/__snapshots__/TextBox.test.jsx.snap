// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox should render correctly 1`] = `
<input
  data-radium={true}
  onChange={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "background": "#eee",
      "border": "none",
      "borderRadius": "1rem",
      "boxSizing": "border-box",
      "color": "#000",
      "filter": "drop-shadow(2px 5px 2px #0003)",
      "flex": 1,
      "fontFamily": "-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
      "fontSize": "17px",
      "fontStyle": "normal",
      "fontWeight": "400",
      "height": "40px",
      "padding": "0px 10px 0px 10px",
      "width": "100%",
    }
  }
  type="text"
/>
`;
